beacon-chain/blockchain/receive_block.go

```diff
@@ -122,6 +122,15 @@ func (s *Service) ReceiveBlock(ctx context.Context, block interfaces.ReadOnlySig
 	if err := eg.Wait(); err != nil {
 		return err
 	}
+
+	if features.Get().EnablePeerDAS {
+		// We need to save the block before checking data availability since reconstruction needs it.
+		// It is safe to save the block multiple times, since it won't be re-stored if it is already in the cache.
+		if err := s.cfg.BeaconDB.SaveBlock(ctx, blockCopy); err != nil {
```

Comment:
We definitely shouldn't save the block here, as a general rule we only save blocks which are fully verified. Saving an unavailable block will cause issues here.

Reply:
Fixed in 8e33539f0165664e8227663c5fc0b8124d4811e3.